sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-supplier.service.ts

```diff
@@ -0,0 +1,236 @@
+   * @param parentProcessSummary parent process summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @returns a number of update records.
+   */
+  async executeCASIntegrationProcess(
+    parentProcessSummary: ProcessSummary,
+    casSuppliers: CASSupplier[],
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    const summary = new ProcessSummary();
+    parentProcessSummary.children(summary);
+    try {
+      summary.info("Logging on CAS...");
+      const auth = await this.casService.logon();
+      summary.info("Logon successful.");
+      suppliersUpdated = await this.requestCASAndUpdateSuppliers(
+        casSuppliers,
+        summary,
+        auth,
+      );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError && error.name === CAS_AUTH_ERROR) {
+        summary.error(error.message);
+      } else {
+        summary.error("Unexpected error.", error);
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * For each pending CAS supplier, request supplier information to CAS API and update local table.
+   * @param casSuppliers pending CAS suppliers.
+   * @param summary log summary.
+   * @param auth CAS auth details.
+   * @returns a number of update records.
+   */
+  private async requestCASAndUpdateSuppliers(
+    casSuppliers: CASSupplier[],
+    parentProcessSummary: ProcessSummary,
+    auth: CASAuthDetails,
+  ): Promise<number> {
+    let suppliersUpdated = 0;
+    for (const casSupplier of casSuppliers) {
+      const summary = new ProcessSummary();
+      parentProcessSummary.children(summary);
+      summary.info(`Requesting info for CAS supplier id ${casSupplier.id}.`);
+      let supplierResponse = null;
+      try {
+        supplierResponse = await this.casService.getSupplierInfoFromCAS(
+          auth.access_token,
+          casSupplier.student.sinValidation.sin,
+          casSupplier.student.user.lastName.toUpperCase(),
+        );
+        suppliersUpdated = await this.updateActiveAddress(
+          supplierResponse,
+          summary,
+          casSupplier,
+          suppliersUpdated,
+        );
+      } catch (error: unknown) {
+        summary.error(
+          "Error while requesting or updating CAS suppliers.",
+          error,
+        );
+      }
+    }
+    return suppliersUpdated;
+  }
+
+  /**
+   * Updates supplier if finds an item from the response with an active address.
+   * @param supplierResponse CAS supplier response.
+   * @param summary log summary.
+   * @param casSuppliers pending CAS suppliers.
+   * @param suppliersUpdated
+   * @returns a number of update records.
+   */
+  private async updateActiveAddress(
+    supplierResponse: CASSupplierResponse,
+    summary: ProcessSummary,
+    casSupplier: CASSupplier,
+    suppliersUpdated: number,
+  ) {
+    if (supplierResponse.items.length) {
+      const [supplierInfo] = supplierResponse.items;
+      const isThereAnActiveSupplierAddress =
```

Comment:
`isThereAnActiveSupplierAddress` can be misleading because pass the idea that it is a boolean variable.
I would recommend `casSupplier`.